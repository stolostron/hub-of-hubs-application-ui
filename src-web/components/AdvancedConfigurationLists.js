/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'

import React from 'react'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import ResourceList from './common/ResourceList'
import { RESOURCE_TYPES } from '../../lib/shared/constants'
import getResourceDefinitions from '../definitions'
import { makeGetVisibleTableItemsSelector } from '../reducers/common'
import {
  getSelectedId,
  default as QuerySwitcher
} from './common/QuerySwitcher'
import msgs from '../../nls/platform.properties'
import { withLocale } from '../providers/LocaleProvider'
import { AcmTablePaginationContextProvider } from '@stolostron/ui-components'

const AdvancedConfigurationLists = ({
  secondaryHeaderProps,
  location,
  locale
}) => {
  const defaultOption = 'subscriptions'
  const options = [
    { id: 'subscriptions', resourceType: RESOURCE_TYPES.QUERY_SUBSCRIPTIONS },
    { id: 'placementrules', resourceType: RESOURCE_TYPES.QUERY_PLACEMENTRULES },
    { id: 'channels', resourceType: RESOURCE_TYPES.QUERY_CHANNELS }
  ]

  const selectedId = getSelectedId({ location, options, defaultOption })

  const resourceType = options.find(o => o.id === selectedId).resourceType
  const staticResourceData = getResourceDefinitions(resourceType)
  const getVisibleResources = makeGetVisibleTableItemsSelector(resourceType)

  return (
    <AcmTablePaginationContextProvider localStorageKey="advanced-tables-pagination">
      <ResourceList
        key={selectedId}
        tabs={secondaryHeaderProps.tabs}
        title={secondaryHeaderProps.title}
        resourceType={resourceType}
        staticResourceData={staticResourceData}
        getVisibleResources={getVisibleResources}
      >
        {[
          <QuerySwitcher
            key="switcher"
            options={options.map(({ id }) => ({
              id,
              contents: msgs.get(`resource.${id}`, locale)
            }))}
            defaultOption={defaultOption}
          />
        ]}
      </ResourceList>
    </AcmTablePaginationContextProvider>
  )
}

AdvancedConfigurationLists.propTypes = {
  locale: PropTypes.string,
  location: PropTypes.object,
  secondaryHeaderProps: PropTypes.object
}

export default withLocale(withRouter(AdvancedConfigurationLists))
