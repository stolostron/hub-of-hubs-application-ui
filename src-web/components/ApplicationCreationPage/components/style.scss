/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2020. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
@import 'scss/variables';

.creation-view-controls-labels {
  padding: 0 0.3125rem 1.125rem 0 !important;
}

.shared-resource-warning {
  display: inline-flex;
  color: $color__blue-90;
  line-height: 2rem;

  svg {
    fill: #e6ae3c;
    width: 18px;
    height: 18px;
  }

  p {
    font-size: 14px;
    margin-top: 0.5rem;
  }

  p:first-of-type {
    margin-top: 0rem;
  }

  div:first-of-type {
    margin-right: 5px;
  }

  .warning-font {
    font-weight: 600;
  }

  .code-font {
    font-size: 12px;
    font-family: Menlo, Monaco, "Courier New", monospace;
  }
}

.timeWindow-container {
  font-size: 14px;
  position: relative;

  // Active/Block mode btns
  .timeWindow-mode-container {
    display: inline-grid;
    margin-bottom: 10px;
  }

  .mode-btn {
    margin: 5px 0;
  }

  // Time window config - Days of the week section
  .config-days-section {
    margin-bottom: 20px;
  }

  .config-title-required {
    font-weight: 400;
    color: #c9190b;
    display: inline;
    margin-left: 0.25rem;
  }

  .config-title {
    font-weight: 600;
    font-size: 14px;
    margin-bottom: 10px;
  }

  .time-picker-title {
    display: flex;

    .config-title {
      padding-right: 71px;
    }
  }

  .config-descr {
    font-size: 14px;
    margin-bottom: 10px;
  }

  .config-days-selector {
    width: 50%;
    display: grid;
    grid-template-columns: auto auto;

    .first-col {
      grid-column-start: 1;
    }

    .second-col {
      grid-column-start: 2;
    }
  }

  // Time window config - Timezone section
  .config-timezone-section {
    margin-bottom: 20px;
  }

  .config-timezone-combo-box {
    width: 290px;
    background-color: $acm_background_color;
  }

  // Time window config - Start/End time section
  .config-time-section {
    display: block;
  }

  .config-time-container {
    display: flex;
    margin-bottom: 20px;
  }

  .config-input-time {
    float: left;
    margin-right: 10px;
  }

  .add-time-btn {
    width: 220px;
    height: 40px;
    padding: 5px;
    text-align: center;
    cursor: pointer;
  }

  .btn-disabled {
    cursor: not-allowed !important;
  }

  .add-time-btn-icon {
    margin-left: 7px;
    float: left;
  }

  .add-time-btn-text {
    font-weight: 600;
    font-size: 14px;
    color: #06c;
  }

  .remove-time-btn {
    padding: 12px 5px;
    cursor: pointer;
  }
}
