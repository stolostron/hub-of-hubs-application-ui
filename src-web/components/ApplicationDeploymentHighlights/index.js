/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project

import React from 'react'

import { connect } from 'react-redux'
import ApplicationDeploymentHighlightsTerminology from './ApplicationDeploymentHighlightsTerminology'

/* eslint-disable react/prop-types */

const mapStateToProps = state => {
  const { QueryApplicationList, secondaryHeader } = state
  return {
    QueryApplicationList,
    secondaryHeader
  }
}

class ApplicationDeploymentHighlights extends React.Component {
  render() {
    return (
      <div id="DeploymentHighlights">
        <ApplicationDeploymentHighlightsTerminology />
      </div>
    )
  }
}

export default connect(mapStateToProps)(ApplicationDeploymentHighlights)
