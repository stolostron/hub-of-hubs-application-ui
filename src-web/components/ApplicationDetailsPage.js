/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'

import { withRouter } from 'react-router-dom'
import { RESOURCE_TYPES } from '../../lib/shared/constants'
import { typedResourcePageDetails } from './common/ResourcePage'

export default withRouter(
  typedResourcePageDetails(RESOURCE_TYPES.QUERY_APPLICATIONS, [], [], [])
)
