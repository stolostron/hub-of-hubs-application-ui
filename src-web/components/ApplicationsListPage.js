/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'

import React from 'react'
import { RESOURCE_TYPES } from '../../lib/shared/constants'
import { typedResourcePageList } from './common/ResourcePage'
import { AcmTablePaginationContextProvider } from '@stolostron/ui-components'
import { PageSection } from '@patternfly/react-core'

const TypedResourcePageList = typedResourcePageList(
  RESOURCE_TYPES.QUERY_APPLICATIONS,
  [],
  [],
  []
)

const ApplicationsListPage = props => (
  <PageSection>
    <AcmTablePaginationContextProvider localStorageKey="application-table-pagination">
      <TypedResourcePageList {...props} />
    </AcmTablePaginationContextProvider>
  </PageSection>
)

export default ApplicationsListPage
