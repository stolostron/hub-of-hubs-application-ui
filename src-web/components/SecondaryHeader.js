/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'

import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Button } from '@patternfly/react-core'
import {
  AcmPageHeader,
  AcmSecondaryNav,
  AcmSecondaryNavItem
} from '@stolostron/ui-components'
import resources from '../../lib/shared/resources'
import { withRouter, Link } from 'react-router-dom'
import msgs from '../../nls/platform.properties'
import loadable from '@loadable/component'

const AutoRefreshSelect = loadable(() =>
  import(/* webpackChunkName: "autoRefreshSelect" */ './common/AutoRefreshSelect')
)

resources(() => {
  require('../../scss/secondary-header.scss')
})

export class SecondaryHeader extends React.Component {
  constructor(props) {
    super(props)
    this.renderTabs = this.renderTabs.bind(this)
    this.renderLinks = this.renderLinks.bind(this)
  }

  render() {
    const {
      tabs,
      title,
      breadcrumbItems,
      links,
      label,
      labelColor
    } = this.props
    const { locale } = this.context

    const headerArgs = {
      breadcrumb: breadcrumbItems && this.getBreadcrumbs(),
      title: decodeURIComponent(title),
      navigation: tabs &&
        tabs.length > 0 && (
          <AcmSecondaryNav
            aria-label={`${title} ${msgs.get('tabs.label', locale)}`}
          >
            {this.renderTabs()}
          </AcmSecondaryNav>
      ),
      controls: <AutoRefreshSelect route={this.props.location} />,
      actions: (
        <Fragment>
          {links && (
            <div className="secondary-header-links">{this.renderLinks()}</div>
          )}
        </Fragment>
      ),
      switches: (
        <div className="switch-controls">
          <div id="edit-button-portal-id" />
        </div>
      ),
      label,
      labelColor
    }
    return <AcmPageHeader {...headerArgs} />
  }

  getBreadcrumbs() {
    const { breadcrumbItems } = this.props
    return breadcrumbItems
      ? breadcrumbItems.map(breadcrumb => {
        return { text: breadcrumb.label, to: breadcrumb.url }
      })
      : null
  }

  renderLinks() {
    const { links } = this.props,
          { locale } = this.context
    return links.map(link => {
      const {
        id,
        label,
        url,
        kind = 'primary',
        title,
        handleClick = () => this.props.history.push(url)
      } = link
      // if portal, react component will create the button using a portal
      if (kind === 'portal') {
        return !title ? <div key={id} id={id} className="portal" /> : null
      }
      return (
        <Button key={id} id={id} onClick={handleClick} kind={kind}>
          {msgs.get(label, locale)}
        </Button>
      )
    })
  }

  renderTabs() {
    const { tabs, location } = this.props,
          { locale } = this.context
    return tabs.map((tab, idx) => {
      return (
        <AcmSecondaryNavItem
          key={tab.id}
          id={tab.id}
          isActive={(this.getSelectedTab() || 0) === idx}
          onClick={tab.handleClick || undefined}
        >
          <Link
            to={{
              ...location,
              pathname: tab.url,
              state: { tabChange: location }
            }}
            replace={true}
          >
            {msgs.get(tab.label, locale)}
          </Link>
        </AcmSecondaryNavItem>
      )
    })
  }

  getSelectedTab() {
    const { tabs, location } = this.props
    const selectedTab = tabs
      .map((tab, index) => {
        tab.index = index
        return tab
      })
      .filter((tab, index) => {
        if (index === 0) {
          return false
        }
        return location.pathname.startsWith(tab.url)
      })
    return selectedTab[0] && selectedTab[0].index
  }
}

SecondaryHeader.propTypes = {
  breadcrumbItems: PropTypes.array,
  history: PropTypes.object,
  label: PropTypes.string,
  labelColor: PropTypes.string,
  links: PropTypes.array,
  location: PropTypes.object,
  tabs: PropTypes.array,
  title: PropTypes.string,
  tooltip: PropTypes.string
}

SecondaryHeader.contextTypes = {
  locale: PropTypes.string
}

const mapStateToProps = state => {
  return {
    title: state.secondaryHeader.title,
    tabs: state.secondaryHeader.tabs,
    breadcrumbItems: state.secondaryHeader.breadcrumbItems,
    links: state.secondaryHeader.links,
    tooltip: state.secondaryHeader.tooltip,
    role: state.role && state.role.role,
    label: state.secondaryHeader.label,
    labelColor: state.secondaryHeader.labelColor
  }
}

export default withRouter(connect(mapStateToProps)(SecondaryHeader))
