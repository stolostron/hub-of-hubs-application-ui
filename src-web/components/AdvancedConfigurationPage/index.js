/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project

import React from 'react'
import msgs from '../../../nls/platform.properties'
import { connect } from 'react-redux'
import resources from '../../../lib/shared/resources'
import {
  updateSecondaryHeader,
  clearSuccessFinished
} from '../../actions/common'

import ApplicationDeploymentHighlights from '../ApplicationDeploymentHighlights'
import { withRouter } from 'react-router-dom'
import AdvancedConfigurationLists from '../AdvancedConfigurationLists'
import { PageSection, Stack, StackItem } from '@patternfly/react-core'

/* eslint-disable react/prop-types */

resources(() => {
  require('./style.scss')
})

const mapDispatchToProps = dispatch => {
  return {
    clearSuccessFinished: () => clearSuccessFinished(dispatch),
    updateSecondaryHeaderFn: (title, tabs) =>
      dispatch(updateSecondaryHeader(title, tabs, null, null, null, null))
  }
}

class AdvancedConfigurationPage extends React.Component {
  componentDidMount() {
    const {
      secondaryHeaderProps,
      updateSecondaryHeaderFn,
      locale
    } = this.props

    updateSecondaryHeaderFn(
      msgs.get(secondaryHeaderProps.title, locale),
      secondaryHeaderProps.tabs
    )
  }

  componentWillUnmount() {
    this.mutateFinished()
  }

  mutateFinished() {
    this.props.clearSuccessFinished()
  }

  render() {
    const { secondaryHeaderProps } = this.props

    return (
      <PageSection>
        <Stack hasGutter>
          <StackItem>
            <ApplicationDeploymentHighlights />
          </StackItem>
          <StackItem>
            <AdvancedConfigurationLists
              secondaryHeaderProps={secondaryHeaderProps}
            />
          </StackItem>
        </Stack>
      </PageSection>
    )
  }
}

export default withRouter(
  connect(null, mapDispatchToProps)(AdvancedConfigurationPage)
)
