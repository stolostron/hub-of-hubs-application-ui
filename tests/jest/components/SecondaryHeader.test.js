/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
"use strict";

import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import { SecondaryHeader } from "../../../src-web/components/SecondaryHeader";

describe("SecondaryHeader component 1", () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });
  it("renders as expected with window size < 1200", () => {
    window.innerWidth = 1000;
    const component = renderer.create(
      <BrowserRouter>
        <SecondaryHeader title="hello world" />
      </BrowserRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("renders as expected with window size > 1200", () => {
    window.innerWidth = 1300;
    const component = renderer.create(
      <BrowserRouter>
        <SecondaryHeader title="hello world" />
      </BrowserRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("SecondaryHeader component 2", () => {
  const tabs = [
    {
      id: "dashboard-application",
      label: "tabs.dashboard.application",
      url: "/multicloud/dashboard"
    }
  ];
  it("renders as expected", () => {
    const component = renderer.create(
      <BrowserRouter>
        <SecondaryHeader title="hello world" tabs={tabs} />
      </BrowserRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("SecondaryHeader component 3", () => {
  const tabs = [
    {
      id: "logs-tab1",
      label: "tabs.dashboard.application",
      url: "/multicloud/dashboard"
    },
    {
      id: "logs-tab2",
      label: "tabs.dashboard",
      url: "/hello"
    }
  ];
  const location = {
    pathname: "/hello"
  };
  it("renders as expected", () => {
    const component = renderer.create(
      //eslint-disable-next-line
      <BrowserRouter>
        <SecondaryHeader title="hello world" tabs={tabs} location={location} />
      </BrowserRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
