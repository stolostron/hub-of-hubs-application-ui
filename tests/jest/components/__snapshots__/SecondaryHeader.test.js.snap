// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryHeader component 1 renders as expected with window size < 1200 1`] = `
<section
  className="pf-c-page__main-section pf-m-no-padding pf-m-light"
>
  <div
    className="pf-l-split"
  >
    <div
      className="pf-l-split__item pf-m-fill"
    >
      <div
        className="pf-l-stack pf-m-gutter"
      >
        <div
          className="pf-l-stack__item pf-m-fill"
        >
          <section
            className="pf-c-page__main-section pf-m-light"
            style={
              Object {
                "paddingBottom": undefined,
                "paddingTop": undefined,
              }
            }
          >
            <div
              className="pf-l-stack pf-m-gutter"
            >
              <div
                className="pf-l-stack__item pf-m-fill"
              >
                <div
                  className="pf-l-split pf-m-gutter"
                >
                  <div
                    className="pf-l-split__item"
                  >
                    <div
                      className="pf-c-content"
                    >
                      <h1
                        className="pf-c-title pf-m-2xl"
                        data-ouia-component-id="OUIA-Generated-Title-1"
                        data-ouia-component-type="PF4/Title"
                        data-ouia-safe={true}
                      >
                        hello world
                      </h1>
                    </div>
                  </div>
                  <div
                    className="pf-l-split__item"
                  >
                    <span
                      style={
                        Object {
                          "alignItems": "center",
                          "display": "flex",
                          "height": "100%",
                          "paddingLeft": "24px",
                        }
                      }
                    >
                      <div
                        className="switch-controls"
                      >
                        <div
                          id="edit-button-portal-id"
                        />
                      </div>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
    <div
      className="pf-l-split__item"
    >
      <section
        className="pf-c-page__main-section pf-m-light"
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <div
          className="pf-l-stack pf-m-gutter"
        >
          <div
            className="pf-l-stack__item"
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
                "flexDirection": "column",
              }
            }
          />
          <div
            className="pf-l-stack__item pf-m-fill"
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "flex-end",
              }
            }
          />
        </div>
      </section>
    </div>
  </div>
</section>
`;

exports[`SecondaryHeader component 1 renders as expected with window size > 1200 1`] = `
<section
  className="pf-c-page__main-section pf-m-no-padding pf-m-light"
>
  <div
    className="pf-l-split"
  >
    <div
      className="pf-l-split__item pf-m-fill"
    >
      <div
        className="pf-l-stack pf-m-gutter"
      >
        <div
          className="pf-l-stack__item pf-m-fill"
        >
          <section
            className="pf-c-page__main-section pf-m-light"
            style={
              Object {
                "paddingBottom": undefined,
                "paddingTop": undefined,
              }
            }
          >
            <div
              className="pf-l-stack pf-m-gutter"
            >
              <div
                className="pf-l-stack__item pf-m-fill"
              >
                <div
                  className="pf-l-split pf-m-gutter"
                >
                  <div
                    className="pf-l-split__item"
                  >
                    <div
                      className="pf-c-content"
                    >
                      <h1
                        className="pf-c-title pf-m-2xl"
                        data-ouia-component-id="OUIA-Generated-Title-2"
                        data-ouia-component-type="PF4/Title"
                        data-ouia-safe={true}
                      >
                        hello world
                      </h1>
                    </div>
                  </div>
                  <div
                    className="pf-l-split__item"
                  >
                    <span
                      style={
                        Object {
                          "alignItems": "center",
                          "display": "flex",
                          "height": "100%",
                          "paddingLeft": "24px",
                        }
                      }
                    >
                      <div
                        className="switch-controls"
                      >
                        <div
                          id="edit-button-portal-id"
                        />
                      </div>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
    <div
      className="pf-l-split__item"
    >
      <section
        className="pf-c-page__main-section pf-m-light"
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <div
          className="pf-l-stack pf-m-gutter"
        >
          <div
            className="pf-l-stack__item"
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
                "flexDirection": "column",
              }
            }
          />
          <div
            className="pf-l-stack__item pf-m-fill"
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "flex-end",
              }
            }
          />
        </div>
      </section>
    </div>
  </div>
</section>
`;

exports[`SecondaryHeader component 2 renders as expected 1`] = `
<section
  className="pf-c-page__main-section pf-m-no-padding pf-m-light"
>
  <div
    className="pf-l-split"
  >
    <div
      className="pf-l-split__item pf-m-fill"
    >
      <div
        className="pf-l-stack pf-m-gutter"
      >
        <div
          className="pf-l-stack__item pf-m-fill"
        >
          <section
            className="pf-c-page__main-section pf-m-light"
            style={
              Object {
                "paddingBottom": "inherit",
                "paddingTop": undefined,
              }
            }
          >
            <div
              className="pf-l-stack pf-m-gutter"
            >
              <div
                className="pf-l-stack__item pf-m-fill"
              >
                <div
                  className="pf-l-split pf-m-gutter"
                >
                  <div
                    className="pf-l-split__item"
                  >
                    <div
                      className="pf-c-content"
                    >
                      <h1
                        className="pf-c-title pf-m-2xl"
                        data-ouia-component-id="OUIA-Generated-Title-3"
                        data-ouia-component-type="PF4/Title"
                        data-ouia-safe={true}
                      >
                        hello world
                      </h1>
                    </div>
                  </div>
                  <div
                    className="pf-l-split__item"
                  >
                    <span
                      style={
                        Object {
                          "alignItems": "center",
                          "display": "flex",
                          "height": "100%",
                          "paddingLeft": "24px",
                        }
                      }
                    >
                      <div
                        className="switch-controls"
                      >
                        <div
                          id="edit-button-portal-id"
                        />
                      </div>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
        <div
          className="pf-l-stack__item"
        >
          <section
            className="pf-c-page__main-nav pf-m-light"
            style={
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <nav
              aria-label="Local"
              className="pf-c-nav pf-m-horizontal pf-m-tertiary"
              data-ouia-component-id="OUIA-Generated-Nav-tertiary-1"
              data-ouia-component-type="PF4/Nav"
              data-ouia-safe={true}
            >
              <button
                aria-label="Scroll left"
                className="pf-c-nav__scroll-button"
                disabled={false}
                onClick={[Function]}
                tabIndex={null}
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                  />
                </svg>
              </button>
              <ul
                className="pf-c-nav__list"
                onScroll={[Function]}
              >
                <li
                  className="pf-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-1"
                  data-ouia-component-type="PF4/NavItem"
                  data-ouia-safe={true}
                  onMouseOver={[Function]}
                >
                  <a
                    aria-current="page"
                    className="pf-c-nav__link pf-m-current"
                    href="/multicloud/dashboard"
                    onClick={[Function]}
                    tabIndex={null}
                  >
                    Application
                  </a>
                </li>
              </ul>
              <button
                aria-label="Scroll right"
                className="pf-c-nav__scroll-button"
                disabled={false}
                onClick={[Function]}
                tabIndex={null}
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                  />
                </svg>
              </button>
            </nav>
          </section>
        </div>
      </div>
    </div>
    <div
      className="pf-l-split__item"
    >
      <section
        className="pf-c-page__main-section pf-m-light"
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <div
          className="pf-l-stack pf-m-gutter"
        >
          <div
            className="pf-l-stack__item"
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
                "flexDirection": "column",
              }
            }
          />
          <div
            className="pf-l-stack__item pf-m-fill"
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "flex-end",
              }
            }
          />
        </div>
      </section>
    </div>
  </div>
</section>
`;

exports[`SecondaryHeader component 3 renders as expected 1`] = `
<section
  className="pf-c-page__main-section pf-m-no-padding pf-m-light"
>
  <div
    className="pf-l-split"
  >
    <div
      className="pf-l-split__item pf-m-fill"
    >
      <div
        className="pf-l-stack pf-m-gutter"
      >
        <div
          className="pf-l-stack__item pf-m-fill"
        >
          <section
            className="pf-c-page__main-section pf-m-light"
            style={
              Object {
                "paddingBottom": "inherit",
                "paddingTop": undefined,
              }
            }
          >
            <div
              className="pf-l-stack pf-m-gutter"
            >
              <div
                className="pf-l-stack__item pf-m-fill"
              >
                <div
                  className="pf-l-split pf-m-gutter"
                >
                  <div
                    className="pf-l-split__item"
                  >
                    <div
                      className="pf-c-content"
                    >
                      <h1
                        className="pf-c-title pf-m-2xl"
                        data-ouia-component-id="OUIA-Generated-Title-4"
                        data-ouia-component-type="PF4/Title"
                        data-ouia-safe={true}
                      >
                        hello world
                      </h1>
                    </div>
                  </div>
                  <div
                    className="pf-l-split__item"
                  >
                    <span
                      style={
                        Object {
                          "alignItems": "center",
                          "display": "flex",
                          "height": "100%",
                          "paddingLeft": "24px",
                        }
                      }
                    >
                      <div
                        className="switch-controls"
                      >
                        <div
                          id="edit-button-portal-id"
                        />
                      </div>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
        <div
          className="pf-l-stack__item"
        >
          <section
            className="pf-c-page__main-nav pf-m-light"
            style={
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              }
            }
          >
            <nav
              aria-label="Local"
              className="pf-c-nav pf-m-horizontal pf-m-tertiary"
              data-ouia-component-id="OUIA-Generated-Nav-tertiary-2"
              data-ouia-component-type="PF4/Nav"
              data-ouia-safe={true}
            >
              <button
                aria-label="Scroll left"
                className="pf-c-nav__scroll-button"
                disabled={false}
                onClick={[Function]}
                tabIndex={null}
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                  />
                </svg>
              </button>
              <ul
                className="pf-c-nav__list"
                onScroll={[Function]}
              >
                <li
                  className="pf-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-2"
                  data-ouia-component-type="PF4/NavItem"
                  data-ouia-safe={true}
                  onMouseOver={[Function]}
                >
                  <a
                    aria-current={null}
                    className="pf-c-nav__link"
                    href="/multicloud/dashboard"
                    onClick={[Function]}
                    tabIndex={null}
                  >
                    Application
                  </a>
                </li>
                <li
                  className="pf-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-3"
                  data-ouia-component-type="PF4/NavItem"
                  data-ouia-safe={true}
                  onMouseOver={[Function]}
                >
                  <a
                    aria-current="page"
                    className="pf-c-nav__link pf-m-current"
                    href="/hello"
                    onClick={[Function]}
                    tabIndex={null}
                  />
                </li>
              </ul>
              <button
                aria-label="Scroll right"
                className="pf-c-nav__scroll-button"
                disabled={false}
                onClick={[Function]}
                tabIndex={null}
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                  />
                </svg>
              </button>
            </nav>
          </section>
        </div>
      </div>
    </div>
    <div
      className="pf-l-split__item"
    >
      <section
        className="pf-c-page__main-section pf-m-light"
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <div
          className="pf-l-stack pf-m-gutter"
        >
          <div
            className="pf-l-stack__item"
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
                "flexDirection": "column",
              }
            }
          />
          <div
            className="pf-l-stack__item pf-m-fill"
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "flex-end",
              }
            }
          />
        </div>
      </section>
    </div>
  </div>
</section>
`;
